en/guide/define_xml_element.md

```diff
@@ -0,0 +1,205 @@
+   - takes its version from **common.xml**  (which is why the `version` tags are commented out).
+
+1. Update the includes:
+   - if the dialect is not based on **common.xml** remove the existing `include` line
+   - Add additional `<include> </include>` elements if you want to import additional files/dialects.
+     > **Note** Includes in nested files are ignored.
+1. Update the version:
+   - if the dialect is not based on **common.xml** uncomment the `<version>6</version>` line and use whatever version you like.
+       > **Note** The version specified in the current file is used by default. If it is not present then the version in included files is used. Most dialects inherit the version from **common.xml**.
+1. Update the `<dialect>8</dialect>` line to replace `8` with the next-largest unused dialect number (based on the other files in the folder).
+1. Optionally remove the `enums` or `messages` sections if you don't plan on declaring any of these elements.
+1. Add enums or messages as described in the following sections.
+1. Save the file, and create a PR to push it back to the **mavlink/mavlink** project repo.
+
+
+
+## Enums
+
+TBD
+
+<!-- 
+Assumptions
+- You can add new enums without risk provided they have a unique name
+- You can add new entries to an enum provided each has a unique name and value (what if clash)
+- You can add new entries for an existing enum in dialect, provided again that the value/name is unique
+- entry values and numbers must be unique ? - 
+
+Questions
+- If a dialect adds enums, what ranges can it use?
+
+
+```xml
+<enum name="LANDING_TARGET_TYPE">
+    <description>Type of landing target</description>
+    <entry value="0" name="LANDING_TARGET_TYPE_LIGHT_BEACON">
+        <description>Landing target signaled by light beacon (ex: IR-LOCK)</description>
+    </entry>
+    <entry value="1" name="LANDING_TARGET_TYPE_RADIO_BEACON">
+        <description>Landing target signaled by radio beacon (ex: ILS, NDB)</description>
+    </entry>
+    <entry value="2" name="LANDING_TARGET_TYPE_VISION_FIDUCIAL">
+        <description>Landing target represented by a fiducial marker (ex: ARTag)</description>
+    </entry>
+    <entry value="3" name="LANDING_TARGET_TYPE_VISION_OTHER">
+        <description>Landing target represented by a pre-defined visual shape/feature (ex: X-marker, H-marker, square)</description>
+    </entry>
+```
+-->
+
+
+## Messages
+
+TBD
+
+<!-- 
+Assumptions/Questions
+- You can add new messages without risk of clash provided they have a unique name and id
+  - Q: If common adds new messages
+      - what id ranges can it use in mavlink 1 and mavlink 2
+      - What is allocation strategy? - just next available?
+      - When can/should we add new messages in MAVLink 1?
+  - Q: If a dialect adds new messages, 
+      - what id ranges can it use in mavlink 1 and mavlink 2
+      - What is allocation strategy?
+- There is a maximum size for a message/max number of fields?
```

Comment:
For mavlink 1.0 I think the entire message must be smaller than 256 bytes.

Reply:
OK, so FYI, the payload max size is 255 bytes in both versions (message size is bigger than that). However the generator does not appear to check that.
The maximum number of fields is 64. The generator crashes with an exception for any more.